### ISSUE TITLE
Winget returning 403 forbidden

### ISSUE THREAD

**Issue by Smith8154** [microsoft/winget-cli#3643]
### Brief description of your issue

When I try to run `winget source update`, I am unable to pull any packages from the default "winget" repository. When checking the logs, I see it is returning a 403 error. This is not happening on all of my machines, it only seems to be machines where part-0010.t-0009.t-msedge.net resolves to IP 13.107.246.38.

Here is the output from the logs:
```
2023-09-20 09:58:08.156 [CORE] WinGet, version [1.4.11071], activity [{9B614010-CE96-40B1-AE16-0A23890FAFDA}]
2023-09-20 09:58:08.158 [CORE] OS: Windows.Desktop v10.0.19045.3448
2023-09-20 09:58:08.159 [CORE] Command line Args: winget  source update
2023-09-20 09:58:08.159 [CORE] IsCOMCall:0; Caller: winget-cli
2023-09-20 09:58:08.159 [CLI ] WinGet invoked with arguments: 'source' 'update'
2023-09-20 09:58:08.159 [CLI ] Found subcommand: source
2023-09-20 09:58:08.159 [CLI ] Found subcommand: update
2023-09-20 09:58:08.159 [CLI ] Leaf command to execute: root:source:update
2023-09-20 09:58:08.160 [CLI ] Executing command: update
2023-09-20 09:58:08.167 [REPO] GetCurrentSourceRefs: Source named 'microsoft.builtin.desktop.frameworks' from origin Default is hidden and is dropped.
2023-09-20 09:58:08.187 [REPO] Named source requested, found: msstore
2023-09-20 09:58:08.194 [REPO] Named source to be updated, found: msstore
2023-09-20 09:58:08.299 [REPO] Named source requested, found: winget
2023-09-20 09:58:08.305 [REPO] Named source to be updated, found: winget
2023-09-20 09:58:08.417 [FAIL] WindowsPackageManager.dll!00007FFF27E7C3B2: ReturnHr(1) tid(231c) 80190193 Forbidden (403).

    Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\HttpStream\HttpClientWrapper.cpp(50)\WindowsPackageManager.dll!00007FFF27E624CA: (caller: 00007FFF27D47F19) Exception(1) tid(231c) 80190193 Forbidden (403).

] 

2023-09-20 09:58:08.417 [FAIL] WindowsPackageManager.dll!00007FFF27D04401: LogHr(1) tid(231c) 80190193 Forbidden (403).


2023-09-20 09:58:08.417 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(53)\WindowsPackageManager.dll!00007FFF27E878E1: (caller: 00007FFF27D5237C) LogHr(2) tid(231c) 80190193 Forbidden (403).

    Msg:[winrt::hresult_error: Forbidden (403).] 

2023-09-20 09:58:08.417 [REPO] Source add/update failed, waiting a bit and retrying: winget
2023-09-20 09:58:10.499 [FAIL] WindowsPackageManager.dll!00007FFF27E7C3B2: ReturnHr(2) tid(231c) 80190193 Forbidden (403).

    Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\HttpStream\HttpClientWrapper.cpp(50)\WindowsPackageManager.dll!00007FFF27E624CA: (caller: 00007FFF27D47F19) Exception(2) tid(231c) 80190193 Forbidden (403).

] 

2023-09-20 09:58:10.499 [FAIL] WindowsPackageManager.dll!00007FFF27D04401: LogHr(3) tid(231c) 80190193 Forbidden (403).


2023-09-20 09:58:10.500 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(663)\WindowsPackageManager.dll!00007FFF27E8750E: (caller: 00007FFF27C9005D) LogHr(4) tid(231c) 80190193 Forbidden (403).

    Msg:[winrt::hresult_error: Forbidden (403).] 

2023-09-20 09:58:10.500 [REPO] Failed to update source: winget
2023-09-20 09:58:10.526 [CLI ] Leaf command succeeded: root:source:update
```

### Steps to reproduce

From a machine that resolves to the broken IP, run `winget source update`.

### Expected behavior

I expect winget to be able to update packages from the default repository.

### Actual behavior

The default winget repository is returning a 403 error.

### Environment

```shell
Windows Package Manager v1.4.11071
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.3448
System Architecture: X64

Logs: %TEMP%\WinGet\defaultState

User Settings: %LOCALAPPDATA%\Microsoft\WinGet\Settings\defaultState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by denelon** [comment]
@Smith8154 thanks for reporting this. Can you download the "winget" cache via https://cdn.winget.microsoft.com/cache/source.msix from one of the affected devices?

**Comment by denelon** [comment]
I noticed for the device originally submitted, it's WinGet 1.4. Can you verify if this is happening on different client versions? The latest stable version is WinGet 1.5 and we're getting close to releasing WinGet 1.6 (end of the month).

**Comment by Smith8154** [comment]
With that said, I have other applications pulling from WinGet repositories that are still returning a 403 error (namely Nerdio, which I use to manage our fleet of AVD machines). I have a support ticket open with Nerdio for this, as I am not sure how they interact with WinGet. If functionality for version 1.4 could be fixed, I am betting Nerdio would be able to pull from the repository as well.

**Comment by Smith8154** [comment]
I shared your contact details with my account manager and he is passing it on to development. For now, Nerdio was able to resolve the issue so I am good to go. They seemed excited to have someone at Microsoft to potentially work with, as their WinGet implementation is still very new.

I am potentially fine with closing this issue out, but was whatever change was made intended to break older versions of WinGet? I can see this potentially causing issues for other users in the future.

**Comment by denelon** [comment]
@Smith8154 
We're doing some work to retire the original CDN used for WinGet we have a few more steps with both client changes and service side changes. We had intended to do this in a non-breaking way, but we found an edge case (actually still present in WinGet 1.5). We will be making a couple of additional minor changes to the WinGet 1.6 Release candidate to allow the process to continue after the newer version of the client is released (expected later this month).

We are still investigating if it will be possible to do this without causing failures in the earlier versions of the WinGet client, but we have not found a perfect solution yet. We will make announcements via pinned issues here at GitHub.
